eslint.config.cjs

```diff
@@ -0,0 +1,64 @@
+const mocha = require("eslint-plugin-mocha");
+const globals = require("globals");
+const js = require("@eslint/js");
+
+const {
+    FlatCompat,
+} = require("@eslint/eslintrc");
+
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+
+module.exports = [{
+    ignores: [
+        // this needs to combine all other eslintignore files so that lint-staged knows what to ignore
+        "packages/cli/scaffold/*.js",
+        "packages/cli/src/generators/templates/**/*.template.js",
+        "**/.yarn",
+        "example-apps/onedrive/test/*",
```

Comment:
Note: I added this new ignore rule for onedrive example app tests; there seems to be some issue with eslint trying to read the `.eslintrc.json` in that root directory (?!) which I didn't want to get into...

Reply:
Yikes, yeah that seems reasonable!